pkg/api/validation/validation.go

```diff
@@ -2479,10 +2565,10 @@ func ValidatePod(pod *api.Pod) field.ErrorList {
 func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allVolumes, vErrs := ValidateVolumes(spec.Volumes, fldPath.Child("volumes"))
+	_, vErrs := ValidateVolumes(spec.Volumes, fldPath.Child("volumes"))
```

Comment:
What about changing the return value to a map instead of a set?  Then the lookup later on will also be faster.